aws/src/test/java/nl/nn/adapterframework/filesystem/AmazonS3FileSystemTestHelper.java

```diff
@@ -0,0 +1,203 @@
+
+	protected String accessKey    = PropertyUtil.getProperty("AmazonS3.properties", "accessKey");
+	protected String secretKey    = PropertyUtil.getProperty("AmazonS3.properties", "secretKey");
+
+	protected @Getter String bucketName   = PropertyUtil.getProperty("AmazonS3.properties", "bucketName");
+
+	private Regions clientRegion = Regions.EU_WEST_1;
+	public static final int S3_PORT = 19090;
+	private String serviceEndpoint = "http://localhost:"+S3_PORT;
+//	private String serviceEndpoint = null;
+
+	private @Getter AmazonS3 s3Client;
+
+	public Path tempFolder;
+
+	private S3Mock s3Mock;
+
+	public AmazonS3FileSystemTestHelper(Path tempFolder) {
+		this.tempFolder = tempFolder;
+	}
+
+	@Override
+	public void setUp() throws Exception {
+		if(serviceEndpoint != null) {
+			s3Mock = new S3Mock.Builder().withPort(S3_PORT).withInMemoryBackend().build();
+			s3Mock.start();
+		}
+
+		s3Client = createS3Client();
+
+		if (!s3Client.doesBucketExistV2(bucketName)) {
+			s3Client.createBucket(bucketName);
+		}
+	}
+
+	//For testing purposes
+	private AmazonS3 createS3Client() {
+		BasicAWSCredentials awsCreds = new BasicAWSCredentials(accessKey, secretKey);
+		AmazonS3ClientBuilder s3ClientBuilder = AmazonS3ClientBuilder.standard()
+				.withChunkedEncodingDisabled(false)
+				.withForceGlobalBucketAccessEnabled(false)
+				.withCredentials(new AWSStaticCredentialsProvider(awsCreds))
+				.withClientConfiguration(new ClientConfiguration().withSocketTimeout(1000).withConnectionTimeout(1000))
+				.enablePathStyleAccess();
+
+		if(StringUtils.isBlank(serviceEndpoint)) {
+			s3ClientBuilder.withRegion(clientRegion);
+		} else {
+			s3ClientBuilder.withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration(serviceEndpoint, clientRegion.getName()));
+		}
+
+		return s3ClientBuilder.build();
+	}
+
+	@Override
+	public void tearDown() throws Exception {
+		if(s3Mock != null) {
+			s3Mock.shutdown();
+		}
+	}
+
+	@Override
+	public boolean _fileExists(String folder, String filename) {
+		String objectName;
+		if(filename == null) {
+			objectName = folder;
+		}else {
+			objectName = folder == null ? filename : folder +"/"+ filename;
+		}
+		try {
+			return s3Client.doesObjectExist(bucketName, objectName);
+		} catch(AmazonS3Exception e) {
+			return false;
+		}
+	}
+
+	@Override
+	public void _deleteFile(String folder, String filename) {
+		String filePath = folder == null ? filename : folder +"/" + filename;
+		s3Client.deleteObject(bucketName, filePath);
+	}
+
+	@Override
+	public OutputStream _createFile(final String foldername, final String filename) throws IOException {
+
+		String fileName = tempFolder.toAbsolutePath()+"tempFile";
+
+		final File file = new File(fileName);
+		final FileOutputStream fos = new FileOutputStream(file);
+		final BufferedOutputStream bos = new BufferedOutputStream(fos);
+
+		FilterOutputStream filterOutputStream = new FilterOutputStream(bos) {
```

Comment:
Subclass van `FilterOutputStream` is hier niet nodig, kan direct subclass zijn van de `BufferedOutputStream`.